virtual-desktop/src/app/window-manager/mvd-window-manager/desktop/desktop.component.ts

```diff
@@ -24,22 +27,86 @@ export class DesktopComponent {
 contextMenuDef: {xPos: number, yPos: number, items: ContextMenuItem[]} | null;
 private authenticationManager: MVDHosting.AuthenticationManagerInterface;
 public isPersonalizationPanelVisible: boolean;
+private needLogin: boolean;
+private changePassword: boolean;
+private previousOpenWindows: DesktopWindow[];
 constructor(
     public windowManager: WindowManagerService,
+    private authenticationService: AuthenticationManager,
     private http: Http,
     private injector: Injector
   ) {
     // Workaround for AoT problem with namespaces (see angular/angular#15613)
     this.authenticationManager = this.injector.get(MVDHosting.Tokens.AuthenticationManagerToken);
     this.contextMenuDef = null;
+    this.needLogin = true;
     this.authenticationManager.registerPostLoginAction(new AppDispatcherLoader(this.http));
+    this.previousOpenWindows = []
+    this.authenticationService.loginScreenVisibilityChanged.subscribe((eventReason: LoginScreenChangeReason) => {
+      switch (eventReason) {
+      case LoginScreenChangeReason.UserLogout:
+        this.needLogin = true;
+        this.previousOpenWindows = [];
+        break;
+      case LoginScreenChangeReason.UserLogin:
+        this.needLogin = false;
+        this.previousOpenWindows = [];
+        break;
+      case LoginScreenChangeReason.SessionExpired:
+        this.needLogin = true;
+        break;
+      case LoginScreenChangeReason.PasswordChange:
+        this.hideVisibleApplications();
+        this.changePassword = true;
+        this.hidePersonalizationPanel();
+        break;
+      case LoginScreenChangeReason.PasswordChangeSuccess:
+        const notifTitle = "Account Password";
+        const notifMessage = "Password was successfully changed."
+        const desktopPluginId = ZoweZLUX.pluginManager.getDesktopPlugin().getIdentifier();
+        this.changePassword = false;
+        this.hidePersonalizationPanel();
+        this.showPreviouslyVisibleApplications();
+        ZoweZLUX.notificationManager.notify(ZoweZLUX.notificationManager.createNotification(notifTitle, notifMessage, 1, desktopPluginId));
+        break;
+      case LoginScreenChangeReason.HidePasswordChange:
+        this.changePassword = false;
+        this.showPersonalizationPanel();
+        this.showPreviouslyVisibleApplications();
+        break;
+      default:
+      }
+    });
   }
   ngOnInit(): void {
     this.windowManager.contextMenuRequested.subscribe(menuDef => {
       this.contextMenuDef = menuDef;
     });
   }
 
+  checkLaunchbarVisibility(): boolean {
+    return this.needLogin || this.changePassword;
+  }
+
+  get isLoggedIn(): boolean {
+    return this.authenticationManager.getUsername() != null ? true : false;
+  }
+
+  hideVisibleApplications(): void {
```

Comment:
JUst change login screen to set the wallpaper and stop being transparent, since it avoids messing around with the windows.